src/plugins/vis_builder/public/application/components/data_tab/index.tsx

```diff
@@ -3,19 +3,336 @@
+    ? schemaAggTypes.filter((type: string) => destinationValidAggType.includes(type))
+    : [];
+
+  aggProps.aggConfigs?.createAggConfig({
+    type: allowedAggTypes[0] || destinationValidAggType[0],
+    schema: destinationSchema.name,
+    params: {
+      field: fieldName,
+    },
+  });
+};
+
 export const DataTab = () => {
+  // Field Selector panel States
+  const fieldSelectorGroups = [
+    'preDefinedCountMetric',
+    'categoricalFields',
+    'numericalFields',
+    'metaFields',
+  ];
+
+  // Config panel States
+  const vizType = useVisualizationType();
+  const editingState = useTypedSelector(
+    (state) => state.visualization.activeVisualization?.draftAgg
+  );
+  const schemas = vizType.ui.containerConfig.data.schemas;
+  const {
+    services: {
+      data: {
+        search: { aggs: aggService },
+      },
+    },
+  } = useOpenSearchDashboards<VisBuilderServices>();
+
+  const aggProps = useAggs();
+  const [schemaDisplayStates, setSchemaDisplayStates] = useState(() => {
+    return schemas.all.reduce((acc, schema) => {
+      acc[schema.name] = [];
+      return acc;
+    }, {});
+  });
+  const dispatch = useTypedDispatch();
+
+  useEffect(() => {
+    const newState = schemas.all.reduce((acc, schema) => {
+      acc[schema.name] = [];
+      return acc;
+    }, {});
+    const upDatedState = { ...newState, ...schemaDisplayStates };
+    setSchemaDisplayStates(upDatedState);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [schemas, vizType]);
+
+  const handleFieldSelectorToConfigurationPanelTransition = ({
+    source,
+    destination,
+    combine,
+    draggableId,
+  }: DropResult) => {
+    // destination Schema
+    const destinationSchemaName = destination?.droppableId || combine?.droppableId;
+    const destinationSchema = schemas.all.find((schema) => schema.name === destinationSchemaName);
+    const destinationFieldToCombine = combine?.draggableId;
+
+    const newFieldToAdd = draggableId;
+
+    if (!destinationSchema || !destinationSchemaName) {
+      // Invalid drop target selected
+      return;
+    }
+
+    const destinationFields = schemaDisplayStates[destinationSchemaName];
+
+    if (!combine && destination && 1 + destinationFields.length > destinationSchema?.max) {
+      // Can't Add additional Fields
+      return;
+    }
+
+    // Case 1 we are adding a new field
+
+    createNewAggConfig({
+      fieldName: newFieldToAdd,
+      sourceGroup: source.droppableId,
+      destinationSchema,
+      aggProps,
+      aggService,
+      sourceAgg: null,
+    });
+
+    let updatedAggConfigs = aggProps.aggConfigs?.aggs;
+
+    if (combine) {
+      // remove the previously selected Aggreagtion
+      updatedAggConfigs = updatedAggConfigs?.filter((agg) => agg.id !== destinationFieldToCombine);
+    }
+    if (updatedAggConfigs) {
+      dispatch(updateAggConfigParams(updatedAggConfigs.map((agg) => agg.serialize())));
+    }
+  };
+
+  const handleConfigurationPanelTransition = ({
```

Comment:
nit: should we memorize this computation func with useCallback? same question for `handleFieldSelectorToConfigurationPanelTransition`

Reply:
In my opinion that might not work here. The params that we have are source, destination, combine which have droppableId of the area and an index that to the fieldAggregation i.e being moved around. So ideally for 2 differrent transitions the index value may remain same but the underlying fieldAggreagtion can be different. In those scenarios the memoization can prove to be detrimental.